### PR TITLE
.golangci: update linter rules

### PR DESCRIPTION
**Pull request by ellemouton**
Update linter rules & fix some issues.

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -15,23 +15,36 @@ linters-settings:
...
+    multi-if: true
+
 linters:
   enable-all: true
   disable:
+    # Allow dynamic errors.
+    - goerr113
+
+    # We want to allow short variable names.
+    - varnamelen
```
Comment (ViktorTigerstrom) [inline .golangci.yml:46]: :pray:

**Review by ViktorTigerstrom — COMMENTED** [review]
Thanks a lot for this! looks good to me, but would like to check if we should add some rule exceptions to counter the most recent linter issues in #88:

1. Should we add a godox exception for this?
```
gbn/timeout_manager.go:87: gbn/timeout_manager.go:87: Line contains TODO/BUG/FIXME: "TODO(viktor): In the future, we may want..." (godox)
	// TODO(viktor): In the future, we may want to use this to keep track of
```

2. Should we add an exception for the below as well, as changing the packet to `gbn_test` doesn't work?
```Error: gbn/timeout_manager_test.go:1:9: package should be `gbn_test` instead of `gbn` (testpackage)```

**Comment by ellemouton** [comment]
@ViktorTigerstrom  - added those 👍

**Review by ViktorTigerstrom — APPROVED** [review]
tACK LGTM!

Though I'm just also mentioning a few linter ignores I've added in #87 & #88.
But IMO I do think these should actually be enforced by the linter, and it's correct that we need to ignore them when necessary though. Let me know if you think we should add rules to not enforce them though!

Linter rule to avoid adding random magic numbers
https://github.com/ViktorTigerstrom/lightning-node-connect/blob/95cb4299933ea2d49983cdf6237f140906c6e333/gbn/gbn_conn.go#L575

Linter rule for requiring an if statement rather than a single clause select statement. The reason I've kept it here is that we'll add more cases to the select statement soon:
https://github.com/ViktorTigerstrom/lightning-node-connect/blob/95cb4299933ea2d49983cdf6237f140906c6e333/gbn/timeout_manager.go#L91

Linter rule to avoid complicated functions with too many cases. In this specific case I don't agree that the function is too complicated though, and breaking it into more sub-functions rather hurts the readability rather than improving it.
https://github.com/ViktorTigerstrom/lightning-node-connect/blob/95cb4299933ea2d49983cdf6237f140906c6e333/gbn/queue.go#L301C4-L301C4

**Review by bhandras — APPROVED** [review]
LGTM 🍰

**Pull request merged**